cabal.project

```diff
@@ -170,6 +174,16 @@ source-repository-package
     plutus-tx
     prettyprinter-configurable
 
+source-repository-package
+  type: git
+  location: https://github.com/input-output-hk/smash
```

Comment:
Isn't this a circular dependency? `Smash` depends on `db-sync` which depends on `smash`.

Reply:
We **really** want to avoid circular dependencies.

Reply:
What is the end goal here? I suspect this is not the right first step to achieve that goal.

Reply:
The Jira ticket talks about getting stake pool metadata from a SMASH server. That is not what this PR does.

Reply:
This is not a circular dependency. It uses some parts of `db-sync` for the plugin mechanism and uses most/all of `smash` to enable that functionality inside together with `db-smash`. This is what they wanted, access to stake pool metadata and the ability to switch between filtering in seconds.

Reply:
It may not be a circular code dependency, but it absolutely is a circular build dedendency.

In this PR, you update the `cabal.project` here in `db-sync` to include things from the `smash` repo. Here in `db-sync` the build depends on things like `cardano-ledger-specs`, but smash also depends on that package. That means that when I try to update the `ledger-specs` dependency for `db-sync`, that new `ledger-specs` version may break the version of `smash` specified by the git hash in the `cabal.project` file. 

This will result in a broken build in the `db-sync` repo. This is why we avoid circular package dependencies.